Files.Launcher/Win32API_ContextMenu.cs

```diff
@@ -0,0 +1,425 @@
+                    Debug.WriteLine(ex);
+                    return null;
+                }
+                catch (Exception ex) when (ex is COMException || ex is NotImplementedException)
+                {
+                    // Not every item has an associated verb
+                    return null;
+                }
+                finally
+                {
+                    commandString?.Dispose();
+                }
+            }
+
+            private static Bitmap GetItemBitmap(HBITMAP hbitmap)
+            {
+                var bitmap = GetTransparentBitmap(hbitmap);
+                bitmap.RotateFlip(RotateFlipType.Rotate270FlipNone);
+                return bitmap;
+            }
+
+            private static Bitmap GetTransparentBitmap(HBITMAP hbitmap)
+            {
+                try
+                {
+                    var dibsection = GetObject<BITMAP>(hbitmap);
+                    var bitmap = new Bitmap(dibsection.bmWidth, dibsection.bmHeight, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+                    using var mstr = new NativeMemoryStream(dibsection.bmBits, dibsection.bmBitsPixel * dibsection.bmHeight * dibsection.bmWidth);
+                    for (var x = 0; x < dibsection.bmWidth; x++)
+                        for (var y = 0; y < dibsection.bmHeight; y++)
+                        {
+                            var rgbquad = mstr.Read<RGBQUAD>();
+                            if (rgbquad.rgbReserved != 0)
+                                bitmap.SetPixel(x, y, rgbquad.Color);
+                        }
+                    return bitmap;
+                }
+                catch { }
+                return Image.FromHbitmap((IntPtr)hbitmap);
+            }
+
+            #region IDisposable Support
+            private bool disposedValue = false; // Per rilevare chiamate ridondanti
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!disposedValue)
+                {
+                    if (disposing)
+                    {
+                        // TODO: eliminare lo stato gestito (oggetti gestiti).
+                        if (Items != null)
+                        {
+                            foreach (var si in Items)
+                                (si as IDisposable)?.Dispose();
+                            Items = null;
+                        }
+                    }
+
+                    // TODO: liberare risorse non gestite (oggetti non gestiti) ed eseguire sotto l'override di un finalizzatore.
+                    if (cMenu != null)
+                    {
+                        Marshal.ReleaseComObject(cMenu);
+                        cMenu = null;
+                    }
+
+                    disposedValue = true;
+                }
+            }
+
+            // TODO: eseguire l'override di un finalizzatore solo se Dispose(bool disposing) include il codice per liberare risorse non gestite.
```

Comment:
Can you write a comment in English so other developers can understand it too.

Reply:
Those comments are created automatically :) I'll translate them